halo2/src/mock_prover.rs

```diff
@@ -5,16 +5,17 @@ use super::circuit_builder::analyzed_to_circuit;
 use halo2_proofs::{dev::MockProver, halo2curves::bn256::Fr};
 use number::{BigInt, FieldElement};
 
+// Can't depend on compiler::pipeline::GeneratedWitness because of circular dependencies...
```

Comment:
on crates or modules?

Reply:
I actually haven't tried, but I don't think the `halo2` crate should depend on `compiler`, because then it would pretty much depend on everything, no?

Reply:
yea we shouldn't do that